### PR TITLE
Use AllPermissionCollection when there is no security manager.

### PR DESCRIPTION
**Pull request by tjwatson**
This avoids trying to add permissions to the collection from the Policy.  That is no longer allowed on Java 24.  There is no need to use the Policy if there is no security manager available.

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Comment by tjwatson** [comment]
#build

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/AbstractJSPExtensionServletWrapper.java [hunk 1]
```diff
@@ -696,6 +676,26 @@ protected void createClassLoader() {
...
     }
 
+    private static final PermissionCollection ALLPERMISSIONS;
+    static {
+        AllPermission allPerm = new AllPermission();
+        ALLPERMISSIONS = allPerm.newPermissionCollection();
+        if (ALLPERMISSIONS != null) {
+            ALLPERMISSIONS.add(allPerm);
+        }
+    }
```
Comment (gjwatts) [inline dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/AbstractJSPExtensionServletWrapper.java:686]: Consider moving static code to the top of the class with the rest.

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/AbstractJSPExtensionServletWrapper.java [hunk 2]
```diff
@@ -617,28 +618,7 @@ protected void createClassLoader() {
...
-                    URL webinfURL = new URL(codeSource.getLocation() + "/WEB-INF/classes/*");
-                    CodeSource webinfCS = new CodeSource(webinfURL, null);
-                    permissionCollection = ((DynamicPolicy) policy).getPermissions(webinfCS, csPerms);
-                }
-            }
-            */
-
-            String sourceDir = jspResources.getGeneratedSourceFile().getParentFile().toString() + File.separator + "*";
-            permissionCollection.add(new FilePermission(sourceDir, "read"));
+            PermissionCollection permissionCollection = createPermissionCollection0();
```
Comment (gjwatts) [inline dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/AbstractJSPExtensionServletWrapper.java:621]:
```suggestion
            PermissionCollection permissionCollection;
            if (System.getSecurityManager() == null) {
                // No need to do anything else when there is no security manager.
                // This handles cases where the security manager isn't supported (e.g. Java 24+).
                permissionCollection = ALLPERMISSIONS;
            } else {
                permissionCollection = createPermissionCollection();
                String sourceDir = jspResources.getGeneratedSourceFile().getParentFile().toString() + File.separator + "*";
                permissionCollection.add(new FilePermission(sourceDir, "read"));
            }
```

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/AbstractJSPExtensionServletWrapper.java [hunk 3]
```diff
@@ -696,6 +676,26 @@ protected void createClassLoader() {
...
+            // No need to do anything else when there is no security manager.
+            // This handles cases where the security manager isn't supported (e.g. Java 24).
+            return ALLPERMISSIONS;
+        }
+        PermissionCollection permissionCollection = createPermissionCollection();
+        String sourceDir = jspResources.getGeneratedSourceFile().getParentFile().toString() + File.separator + "*";
+        permissionCollection.add(new FilePermission(sourceDir, "read"));
+        return permissionCollection;
+    }
+
```
Comment (gjwatts) [inline dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/AbstractJSPExtensionServletWrapper.java:698]:
```suggestion
```

**Comment by tjwatson** [comment]
Moved this commit over to PR #30226

Closing this PR

**Pull request closed without merging**